tutorials/time-freq/plot_ssvep.py

```diff
@@ -0,0 +1,780 @@
+#
+# - Raw data come with FCz recording reference, so we will apply common-average
+# rereferencing.
+# - We will apply a 50 Hz notch filter to remove line noise,
+# - and a 0.1 - 250 Hz bandpass filter.
+# - Lastly, we will cut the data in 20 s epochs corresponding to the trials.
+
+# Load raw data
+data_path = mne.datasets.ssvep.data_path()
+bids_path = BIDSPath(subject='02', session='01', task='ssvep', root=data_path)
+
+# read_raw_bids issues warnings about missing electrodes.tsv and
+# coordsystem.json.
+# These warning prevent successful building of the tutorial.
+# As a quick workaround, we just suppress the warnings here.
+with warnings.catch_warnings():
+    warnings.simplefilter("ignore")
+    raw = read_raw_bids(bids_path, verbose=False)
+raw.load_data()
+
+# Set montage
+montage_style = 'easycap-M1'
+montage = mne.channels.make_standard_montage(
+    montage_style,
+    head_size=0.095)  # head_size parameter default = 0.095
+raw.set_montage(montage, verbose=False)
+
+# Set common average reference
+raw.set_eeg_reference('average', projection=False, verbose=False)
+
+# Apply notch filter to remove line-noise
+notch = np.arange(raw.info['line_freq'], raw.info['lowpass'] / 2,
+                  raw.info['line_freq'])
+raw.notch_filter(notch, filter_length='auto', phase='zero', verbose=False)
+
+# Apply bandpass filter
+hp = .1
+lp = 250.
+raw.filter(hp, lp, fir_design='firwin', verbose=False)
+
+# Construct epochs
+event_id = {
+    '12hz': 10001,
+    '15hz': 10002
+}
+events, _ = mne.events_from_annotations(raw, verbose=False)
+raw.info["events"] = events
+tmin, tmax = -1., 20.  # in s
+baseline = None
+epochs = mne.Epochs(
+    raw, events=events,
+    event_id=[event_id['12hz'], event_id['15hz']], tmin=tmin,
+    tmax=tmax, baseline=baseline, verbose=False)
+
+###############################################################################
+# Frequency analysis
+# ------------------
+# Now we compute the frequency spectrum of the EEG data.
+# You will already see the peaks at the stimulation frequencies and some of
+# their harmonics, without any further processing.
+#
+# The 'classical' PSD plot will be compared to a plot of the SNR spectrum.
+# SNR will be computed as a ratio of the power in a given frequency bin
+# to the average power in its neighboring bins.
+# This procedure has two advantages over using the raw PSD:
+#
+# * it normalizes the spectrum and accounts for 1/f power decay.
+# * power modulations which are not very narrow band will disappear.
+#
+
+###############################################################################
+# Calculate power spectral density (PSD)
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# We use Welch's method for frequency decomposition, since it is really fast.
+# We chose a frequency resolution of 0.1 hz.
+# You could compare it with, e.g., multitaper to get an impression of the
+# influence on SNR. All the other methods implemented in MNE can be used as
+# well.
+
+tmin = 0.
```

Comment:
From the description above it seems that a change in the stimulus occurs at 0s (a switch between 12 and 15 hz). Im not sure about VSSEPs, but for ASSRs this would elicit a neural change response. So we would typically exclude the first ~1s of the epoch as this will have broadband frequency components generated by the change response. Perhaps it would be informative to plot the average epoch to show if this is the case or not? This is 100% curiosity from me about the differences between visual and auditory responses.

Reply:
this is really a toy dataset: 
i presented checkerboards that invert b/w fields with a fix frequency, either 12 or 15 hz depending on the trial (this elicits a steady-state response). there is also an ITI of a second or so.

yes, i agree that excluding the first second would probably improve the signal quality. i didnt do it with these data, explicitly, but towards the end of the tutorial i simulated shorter trials and, in the very first seconds of the trials snr seems lower